contrib/mergeBenchmarkSets.py

```diff
@@ -82,98 +104,85 @@ def getWitnessResult(witness, verification_result):
     return "result invalid (" + status_from_verification + ")", result.CATEGORY_ERROR
 
 
-def main(argv=None):
-
-    if argv is None:
-        argv = sys.argv
-
-    if len(argv) < 3:
-        sys.exit(
-            "Usage: "
-            + argv[0]
-            + " <results-xml> [<witness-xml>]* [--no-overwrite-status-true].\n"
-        )
-
-    resultFile = argv[1]
-    witnessFiles = []
-    isOverwrite = True
-    for i in range(2, len(argv)):
-        if len(argv) > i and not argv[i].startswith("--"):
-            witnessFiles.append(argv[i])
-        if argv[i] == "--no-overwrite-status-true":
-            isOverwrite = False
-
-    if not os.path.exists(resultFile) or not os.path.isfile(resultFile):
-        sys.exit("File {0} does not exist.".format(repr(resultFile)))
-    resultXML = tablegenerator.parse_results_file(resultFile)
-    witnessSets = []
-    for witnessFile in witnessFiles:
-        if not os.path.exists(witnessFile) or not os.path.isfile(witnessFile):
-            sys.exit("File {0} does not exist.".format(repr(witnessFile)))
-        witnessXML = tablegenerator.parse_results_file(witnessFile)
-        witnessSets.append(getWitnesses(witnessXML))
-
-    for result_tag in resultXML.findall("run"):
-        run = result_tag.get("name")
+def getValidationResult(
+    run, witnessSets, status_from_verification, category_from_verification
+):
+    statusWit, categoryWit = None, None
+    name = run.get("name")
+    for witnessSet in witnessSets:
+        witness = witnessSet.get(name)
+        if witness is None or not len(witness):
```

Comment:
This should be the same as `if not witness`, shouldn't it?

Reply:
Not quite, `not witness` would also be `True` in case the witness were an empty list, `witness is None` guarantees that it is actually the `None` object. Depending on the python implementation there might be different `None` objects though :see_no_evil:.

But I agree that `not witness` is probably less errorprone here.

Reply:
The code here is `if witness is None or not len(witness):` and not only `if witness is None:`. What's the difference to `if not witness:`?

Reply:
`if not witness` is less likely to break, e.g. if you pass an integer.

I am not arguing against that `if not witness` :wink:.

Reply:
`witness` should actually a be an `xml.etree.ElementTree.Element`.
`len(witness)` returns the number of children but I wasn't sure if that would be considered when using `if not witness`.
I have confirmed now that it is indeed safe to use `if not witness` here and changed it accordingly.